plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/BaseSqlServerConnectorTest.java

```diff
@@ -342,6 +339,48 @@ public void testPredicatePushdown()
             assertThat(query("SELECT * FROM " + testTable.getName() + " WHERE long_decimal = 123456789.987654321"))
                     .matches("VALUES (CAST(123.321 AS decimal(9,3)), CAST(123456789.987654321 AS decimal(30, 10)))")
                     .isFullyPushedDown();
+
+            // varchar predicate over join
+            Session joinPushdownEnabled = joinPushdownEnabled(getSession());
+            assertThat(query(joinPushdownEnabled, "SELECT c.name, n.name FROM customer c JOIN nation n ON c.custkey = n.nationkey WHERE n.name = 'POLAND'"))
+                    .isFullyPushedDown();
+
+            // join on varchar columns is not pushed down
+            assertThat(query(joinPushdownEnabled, "SELECT c.name, n.name FROM customer c JOIN nation n ON c.address = n.name"))
+                    .joinIsNotFullyPushedDown();
+        }
+    }
+
+    @Test
+    public void testNoPushdownOnCaseInsensitiveVarcharColumn()
```

Comment:
let's also add a test where we verify no pushdown for the other "inequalities" - `<` and `>`. It would make sure we never mistakenly add them because for > and < we need to know exact collation - not just case-sensitivity.

Reply:
Test cases added